core/text/table/table.odin

```diff
@@ -306,45 +425,58 @@ write_ascii_table :: proc(w: io.Writer, tbl: ^Table) {
 // Renders table according to GitHub Flavored Markdown (GFM) specification
 write_markdown_table :: proc(w: io.Writer, tbl: ^Table) {
 	// NOTE(oskar): Captions or colspans are not supported by GFM as far as I can tell.
+	WIDTH_PROC :: unicode_width_proc
 
 	if tbl.dirty {
-		build(tbl)
+		build(tbl, WIDTH_PROC)
 	}
 
-	for row in 0..<tbl.nr_rows {
-		for col in 0..<tbl.nr_cols {
+	write_row :: proc(w: io.Writer, tbl: ^Table, row: int, alignment: Cell_Alignment = .Left) {
+		for col in 0 ..< tbl.nr_cols {
 			cell := get_cell(tbl, row, col)
 			if col == 0 {
 				io.write_byte(w, '|')
 			}
-			write_text_align(w, tbl.colw[col] - tbl.lpad - tbl.rpad, tbl.lpad, tbl.rpad, cell.text,
-			                 .Center if tbl.has_header_row && row == header_row(tbl) else .Left)
+			write_text_align(w, cell.text, alignment, tbl.lpad, tbl.rpad, tbl.colw[col] - cell.width)
 			io.write_string(w, "|")
 		}
 		io.write_byte(w, '\n')
+	}
 
-		if tbl.has_header_row && row == header_row(tbl) {
-			for col in 0..<tbl.nr_cols {
-				cell := get_cell(tbl, row, col)
-				if col == 0 {
-					io.write_byte(w, '|')
-				}
-				switch cell.alignment {
-				case .Left:
-					io.write_byte(w, ':')
-					write_byte_repeat(w, max(1, tbl.colw[col]-1), '-')
-				case .Center:
-					io.write_byte(w, ':')
-					write_byte_repeat(w, max(1, tbl.colw[col]-2), '-')
-					io.write_byte(w, ':')
-				case .Right:
-					write_byte_repeat(w, max(1, tbl.colw[col]-1), '-')
-					io.write_byte(w, ':')
-				}
+	start := 0
+
+	if tbl.has_header_row {
+		row := header_row(tbl)
+
+		write_row(w, tbl, row, .Center)
+
+		for col in 0 ..< tbl.nr_cols {
```

Comment:
`a..<b` (without surrounding space) is the preferred Odin style.

I'll stop pointing these out now. Please fix them all :)